src/lib/exeRunner.ts

```diff
@@ -26,7 +27,10 @@ export class ExeRunner {
             const stderr = new Array<string>();
 
             this.logger.info(`exe: ${this._exePath}, first arg of ${args.length}: ${args.length ? args[0]: '<none>'}`);
-            const process = spawn(this._exePath, args, { cwd: this.workingDir });
+            const process = spawn(this._exePath, args, {
+                cwd: this.workingDir,
+                env: { "PP_TOOLS_AUTOMATION_AGENT": getAutomationAgent() }
```

Comment:
use Object.assign(), see src/CommandRunner.ts in pp-cli-wrapper PR https://github.com/microsoft/powerplatform-cli-wrapper/pull/25

Reply:
please use the spread syntax as being discussed in the other comment thread. both should use the same technique.